gurobipy-stubs/gurobipy.pyi

```diff
@@ -815,7 +815,29 @@ class MConstr:
     @property
     def size(self) -> int: ...
 
+class MQConstr:
+    QConstrName: str = ...
+    QCRHS: float = ...
+    QCSense: str = ...
+    QCSlack: float = ...
+    def getAttr(self, attrname: str) -> np.ndarray: ...
+    @overload
+    def setAttr(self, attrname: str, value: float) -> None: ...
+    @overload
+    def setAttr(self, attrname: str, value: str) -> None: ...
+    @overload
+    def setAttr(self, attrname: str, value: np.ndarray) -> None: ...
+    def tolist(self) -> List[QConstr]: ...
+    @property
+    def ndim(self) -> int: ...
+    @property
+    def shape(self) -> Tuple[int, ...]: ...
+    @property
+    def size(self) -> int: ...
+
 class MLinExpr:
+    @classmethod
+    def zeros(self, shape: Union[int, Tuple[int, ...]]) -> MLinExpr: ...
```

Comment:
The `Union[int, Tuple[int, ...]])` appears often enough that it warrents its own type alias, maybe `ShapeLike`.  Please check the NumPy type hints, I'm certain that they have something for exactly this case, and we could align with that.

Reply:
No ShapeLike equivalent listed [in numpy docs](https://numpy.org/devdocs/reference/typing.html#), so I wrote an alias `_ShapeLike` into our stubs